packages/react/__tests__/get-label-from-stack-trace.js

```diff
@@ -0,0 +1,560 @@
+    at beginWork (react-dom.development.js?3c4a:18596)
+    at beginWork$1 (react-dom.development.js?3c4a:23179)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Firefox', () => {
+    const stackTrace = `createEmotionProps@webpack-internal:///../../packages/react/dist/emotion-element-6352414e.browser.esm.js:163:40
+jsxDEV@webpack-internal:///../../packages/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.browser.esm.js:35:230
+MyComponent$9@webpack-internal:///./pages/index.js:32:82
+App@webpack-internal:///./pages/index.js:49:17
+renderWithHooks@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:14803:27
+mountIndeterminateComponent@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:17482:13
+beginWork@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:18596:16
+beginWork$1@webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:23179:14`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Safari', () => {
+    // MyComponent$9 does not appear in the stack trace
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/main.chunk.js:866:49
+jsxDEV@http://localhost:3000/static/js/main.chunk.js:1147:247
+App@http://localhost:3000/static/js/main.chunk.js:2290:32
+renderWithHooks@http://localhost:3000/static/js/vendors~main.chunk.js:19152:31
+mountIndeterminateComponent@http://localhost:3000/static/js/vendors~main.chunk.js:21764:32
+beginWork$1@http://localhost:3000/static/js/vendors~main.chunk.js:27384:27`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe('App')
+  })
+
+  test('SSR', () => {
+    const stackTrace = `Error
+    at Object.createEmotionProps (webpack-internal:///../../packages/react/dist/emotion-element-7a9c77b4.cjs.dev.js:175:40)
+    at jsxDEV (webpack-internal:///../../packages/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.cjs.dev.js:22:75)
+    at Object.MyComponent$9 (webpack-internal:///./pages/index.js:31:82)
+    at App (webpack-internal:///./pages/index.js:47:19)
+    at processChild (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3043:14)
+    at resolve (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:2960:5)
+    at ReactDOMServerRenderer.render (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3435:22)
+    at ReactDOMServerRenderer.read (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3373:29)
+    at Object.renderToString (C:/Projects/OSS/emotion/node_modules/react-dom/cjs/react-dom-server.node.development.js:3988:27)
+    at Object.renderPage (C:/Projects/OSS/emotion/node_modules/next/dist/server/render.js:621:45)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+})
+
+test('Chrome: Module.jsx', () => {
```

Comment:
could you add a note above this block mentioning how one reproduce this situation?

Reply:
I never encountered a stack trace containing `Module.` during all of my testing so I'm not sure how this happens. It's complicated because it could depend on stuff like which version of Webpack is used.